exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/join/HashJoinBatch.java

```diff
@@ -998,6 +1000,10 @@ public IterOutcome executeBuildPhase() throws SchemaChangeException {
             : read_right_HV_vector.getAccessor().get(ind); // get the hash value from the HV column
           int currPart = hashCode & spilledState.getPartitionMask();
           hashCode >>>= spilledState.getBitsInMask();
+          // semi-join skips join-key-duplicate rows
+          if ( semiJoin ) {
+
```

Comment:
Empty if statement

Reply:
Thanks. That empty if was a reminder for the followup work on the duplicate elimination; maybe we should leave it there as a reminder (it would be overwritten when the 2nd and 3rd commit would be added).